source/MaterialXGenMdl/MdlShaderGenerator.cpp

```diff
@@ -198,6 +198,9 @@ ShaderPtr MdlShaderGenerator::generate(const string& name, ElementPtr element, G
 
     ShaderPtr shader = createShader(name, element, context);
 
+    // Request fixed floating-point notation for consistency across targets.
```

Comment:
I'm wondering if this should be moved to a new method to set the precision that all generators call into ? Then you don't need to modify this in N different places for N generators if this changes.

Reply:
That's a good thought, though I wonder if there's really any simpler expression of this setup than the current one-liner.  In order to wrap an RAII class, you'd need a second RAII class instantiating the first one, which might just end up being additional indirection without improving clarity or universality across languages.

In the future, one option to consider would be a base ShaderGenerator::generate method that handles common steps, though for now I think it's clearer to have this explicit step in each language, so that developers know that this precision formatting is being invoked here, and can test different precisions locally for custom generators.